src/Sparrow.Server/Strings/StringsExtensions.cs

```diff
@@ -188,6 +189,18 @@ internal static unsafe bool CompareConstantAvx2(ref byte constantRef, byte* ptr,
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static unsafe bool CompareConstantVector128(ref byte constantRef, byte* ptr, int size)
         {
+            if (PlatformSpecific.IsArm)
```

Comment:
Let's add a TODO here that explains that this is temporary work around because of a bug

The `Constant` here needs to be constant time operation?

Reply:
Yes, `CompareConstant` has `ReadOnlySpan` as the input parameter.

Reply:
Added todo. What do you mean by constant time?

Reply:
Constant time means that regardless of the result, it completes in the same time.

Relevant usually for crypto.

See: 
https://github.com/ziglang/zig/issues/1776

Reply:
Oh, ok. I don't believe it has to be time constant but it means `constantRef` must not change (since it's `ReadOnlySpan` extension). cc @redknightlois

Reply:
Why are we worried about a timing attack here? The idea of this compare method is that the size is actually a language literal ( you dont call it with a variable, you call it with an actual size, say 8) that causes for the JIT to write specific packed code specialized to that number. Since this is a temporary workaround over a bug, I see no reason why not to disable this on ARM completely, which is essentially what @maciejaszyk is doing by emitting a byte-by-byte comparison code. The only thing that I would do though is that not execute it if value is smaller than 16 which is essentially handled by the switch at the end.

Reply:
BTW, since we already know that the loop is going to be bigger than 16, why not just use shorts comparisons instead.